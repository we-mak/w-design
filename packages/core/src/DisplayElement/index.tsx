import * as React from "react";
import styled from "styled-components";
import { display } from "styled-system";
import { GlobProps } from "../common/props";

const Base = styled.div`
  ${display}
`;

export interface DisplayProps extends GlobProps {
  display: string[];
}

const DisplayElement: React.FunctionComponent<DisplayProps> = (props) => (
  <Base {...props}>{props.children}</Base>
);

export default DisplayElement;
