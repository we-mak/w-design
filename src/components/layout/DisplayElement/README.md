Responsive Hidden based on default theme breakpoints

| Props | type    | default | description                                                       |
| ----- | ------- | ------- | ----------------------------------------------------------------- |
| xs    | boolean | false   | hide at @media screen and (max-width: 24em)                       |
| sm    | boolean | false   | hide at @media screen and (min-width: 24em) and (max-width: 32em) |
| md    | boolean | false   | hide at @media screen and (min-width: 32em) and (max-width: 48em) |
| lg    | boolean | false   | hide at @media screen and (min-width: 48em) and (max-width: 64em) |
| xl    | boolean | false   | hide at @media screen and (min-width: 64em) and (max-width: 80em) |
| xxl   | boolean | false   | hide at @media screen and (min-width: 80em)                       |

```js
<Container>
  <Flex>
    <Flex.Column width={1} bg="#eaeaea">
      <DisplayElement xs>Hide xs element</Hide>
      <DisplayElement lg>Hide lg element</Hide>
    </Flex.Column>
  </Flex>
</Container>
```
