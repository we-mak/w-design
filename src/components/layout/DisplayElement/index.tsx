import * as React from "react";
import styled from "styled-components";
import { display } from "styled-system";

const Base = styled.div`
  ${display}
`;

interface DisplayProps {
  display: string[];
}

const DisplayElement: React.FunctionComponent<DisplayProps> = props => (
  <Base display={props.display} {...props}>
    {props.children}
  </Base>
);

export default DisplayElement;
