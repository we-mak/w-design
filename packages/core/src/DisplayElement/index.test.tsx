import * as React from "react";
import { render } from "@testing-library/react";
import DisplayElement from "./index";
import { Provider } from "../../";

describe("<DisplayElement/>", () => {
  // Column Style test
  it("should has display none with xs", () => {
    const { container } = render(
      <Provider>
        <DisplayElement display={["block"]} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  // it("should has display none with sm", () => {
  //   const wrapper = renderer.create(<DisplayElement sm />).toJSON();
  //   expect(wrapper).toHaveStyleRule("display", "none", {
  //     media: "screen and (min-width: 24em) and (max-width: 32em)"
  //   });
  // });
});
