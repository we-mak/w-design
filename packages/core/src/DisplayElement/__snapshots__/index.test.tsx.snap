// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayElement/> should has display none with xs 1`] = `
<div>
  <div
    class="sc-bdnxRM ctuJRu"
    display="block"
  />
</div>
`;
